tests/spec/filter/test_drop_with_column.py

```diff
@@ -1,43 +1,28 @@
-from databuilder.query_language import (
-    BoolSeries,
-    Dataset,
-    IdSeries,
-    IntSeries,
-    build_event_table,
-)
+from ..tables import e
 
-from ..helpers import transpose
+table_data = {
+    e: """
+          |  i1 |  b1
+        --+-----+-----
+        1 | 101 |  T
+        1 | 102 |  T
+        1 | 103 |
+        2 | 201 |  T
+        2 | 202 |
+        2 | 203 |  F
+        3 | 301 |  T
+        3 | 302 |  T
+    """,
+}
 
-e = build_event_table(
-    "e",
-    {
-        "patient_id": IdSeries,
-        "i": IntSeries,
-        "b": BoolSeries,
-    },
-)
 
-
-def test_drop_with_column(in_memory_engine):
-    in_memory_engine.setup(
+def test_drop_with_column(spec_test):
+    spec_test(
+        table_data,
+        e.drop(e.b1).i1.sum_for_patient(),
```

Comment:
I wonder if we could document the _motivation_ for `drop`/`take` (as opposed to a filter operation) in the specs, showing that `drop(e.b1)` and `take(~e.b1)` don't partition the table.